### DISCUSSION TITLE
Problems with Airflow 2.10.2

### DISCUSSION THREAD

**Discussion by aipatr** [apache/airflow#42434] [General]
After installing Airflow 2.10.2 I get the following message: 
AttributeError: cffi library '_openssl' has no function, constant or global variable named 'Cryptography_HAS_MEM_FUNCTIONS'. Did you mean: 'Cryptography_HAS_TLSv1_3_FUNCTIONS'?

I updated my running version using https://airflow.apache.org/docs/apache-airflow/stable/installation/installing-from-pypi.html
and performed an airflow db migrate.

Could you support me?
Thanks

**Comment by potiuk** [comment]
Likely something similar to https://github.com/pyca/cryptography/issues/4187 (googled it, so that's all I can help you with - you might also continue searching for similar issues - seems it's not related to airflow but cryptography dependency). Could be you whole python installation is broken and you would need to reinstall it.

**Reply by aipatr** [reply]
it is not broken it looks like the version that airflow is using is not really the right one because there is some reference that is in the old package of cryptography. Could someone please check this?`

**Reply by potiuk** [reply]
Yopu have to look how your Airflow is installed - people had similar erro when their python was broken so likey you have similar issue https://stackoverflow.com/questions/54921269/pip3-and-requests-module-attributeerror-cffi-library-openssl-has-no-function 

You need to diagnose your setup. It's not an airlfow user. 1000s of other people did not have the same problem so it is your environmental issue. 

We have completely no idea how you installed Airlfow - but some general rules - do not use stock python that comes with your OS, but install it via official installation methods from Python. If needs be reinstall everything from the scratch.
